test/nut00_test.exs

```diff
@@ -144,32 +144,103 @@ defmodule BDHKETest do
   end
 
   describe "token V4" do
-    # FIXME this test is still broken
-    test "serializes single keyset" do
-      token =
+    test "deserialize single keyset" do
+      serialized =
+        "cashuBpGF0gaJhaUgArSaMTR9YJmFwgaNhYQFhc3hAOWE2ZGJiODQ3YmQyMzJiYTc2ZGIwZGYxOTcyMTZiMjlkM2I4Y2MxNDU1M2NkMjc4MjdmYzFjYzk0MmZlZGI0ZWFjWCEDhhhUP_trhpXfStS6vN6So0qWvc2X3O4NfM-Y1HISZ5JhZGlUaGFuayB5b3VhbXVodHRwOi8vbG9jYWxob3N0OjMzMzhhdWNzYXQ="
+
+      assert {:ok,
+              %TokenV4{
+                mint: "http://localhost:3338",
+                unit: "sat",
+                memo: "Thank you",
+                token: [
+                  %ProofV4{
+                    keyset_id: "00ad268c4d1f5826",
+                    amount: 1,
+                    secret: "9a6dbb847bd232ba76db0df197216b29d3b8cc14553cd27827fc1cc942fedb4e",
+                    signature:
+                      "038618543ffb6b8695df4ad4babcde92a34a96bdcd97dcee0d7ccf98d472126792"
+                  }
+                ]
+              }} = Cashu.Serializer.V4.deserialize(serialized)
+    end
+
+    test "serializes and deserializes single keyset" do
```

Comment:
Here's a simpler version of this test imo. You don't need to check every field if you compare via equality `==` and not pattern match. Also the test title should be about tokens, not keysets.
```
    test "serializes and deserializes single keyset" do
      proofv4 = %ProofV4{
          keyset_id: "00ad268c4d1f5826",
          amount: 1,
          secret: "9a6dbb847bd232ba76db0df197216b29d3b8cc14553cd27827fc1cc942fedb4e",
          signature: "038618543ffb6b8695df4ad4babcde92a34a96bdcd97dcee0d7ccf98d472126792"
        }

      token =
        TokenV4.new(
          [proofv4],
          "http://localhost:3338",
          "sat",
          "Thank you"
        )

      assert {:ok, serialized} = Cashu.Serializer.V4.serialize(token)
      assert {:ok, token} == Cashu.Serializer.V4.deserialize(serialized)
    end
```